### ISSUE TITLE
Can't draw rectangles with terrain layer enabled

### ISSUE THREAD

**Issue by Avitale34** [wsaunders1014/TerrainLayer#15]
Since 7.5 I have been having an issue with drawing rectangles. When terrain Layer is turned off it works as expected. I am getting this error.

foundry.js:3780 Error: Required field type not present in Drawing.
    at Drawing._validateRequiredField (usr/lib/forgevtt/releases/foundryvtt-0.7.7/dist/database/odm/abstract.js:1)
    at Drawing.validate (usr/lib/forgevtt/releases/foundryvtt-0.7.7/dist/database/odm/abstract.js:1)
    at Function._onCreate (usr/lib/forgevtt/releases/foundryvtt-0.7.7/dist/database/odm/embedded.js:1)
    at async Socket.handleEvent (usr/lib/forgevtt/releases/foundryvtt-0.7.7/dist/sockets.js:1)
_handleError @ foundry.js:3780
(anonymous) @ foundry.js:3761
n.onack @ index.js:83
n.onpacket @ index.js:83
(anonymous) @ index.js:83
n.emit @ index.js:83
n.ondecoded @ index.js:83
(anonymous) @ index.js:83
n.emit @ index.js:83
a.add @ index.js:83
n.ondata @ index.js:83
(anonymous) @ index.js:83
n.emit @ index.js:83
n.onPacket @ index.js:83
(anonymous) @ index.js:83
n.emit @ index.js:83
n.onPacket @ index.js:83
n.onData @ index.js:83
ws.onmessage @ index.js:83
foundry.js:3777 Uncaught (in promise) Error: Required field type not present in Drawing.
    at Drawing._validateRequiredField (usr/lib/forgevtt/releases/foundryvtt-0.7.7/dist/database/odm/abstract.js:1)
    at Drawing.validate (usr/lib/forgevtt/releases/foundryvtt-0.7.7/dist/database/odm/abstract.js:1)
    at Function._onCreate (usr/lib/forgevtt/releases/foundryvtt-0.7.7/dist/database/odm/embedded.js:1)
    at async Socket.handleEvent (usr/lib/forgevtt/releases/foundryvtt-0.7.7/dist/sockets.js:1)

**Comment by LarkinOmega** [comment]
This problems occurs in conjunction with Multilevel Tokens and The Furnace.

**Comment by UnoriginalDM** [comment]
Yes, Larkin is correct. It is the "advanced drawing tools" of furnace and terrain layer creating this issue. I don't have a solution to fix the problem, but I have been using the polygon drawl tool to make rectangles to get around this.
